pkgs/development/tools/build-managers/meson/setup-hook.sh

```diff
@@ -20,7 +20,7 @@ mesonConfigurePhase() {
         -Dauto_features=disabled \
         $mesonFlags"
 
-    mesonFlags="${crossMesonFlags+$crossMesonFlags }--buildtype=${mesonBuildType:-release} $mesonFlags"
+    mesonFlags="${crossMesonFlags+$crossMesonFlags} --buildtype=${mesonBuildType:-release} $mesonFlags"
```

Comment:
yes, this seems needless.
At first glance I thought that the problem is here, in missing space so the command line arguments stuck together and that broke cross-compilation.
It could be left as is or optimized to ```mesonFlags="$crossMesonFlags --buildtype…```

Reply:
Well, it does not miss space – `${varname+text}` [expands](http://wiki.bash-hackers.org/syntax/pe#use_an_alternate_value) to `text` when the variable `varname` is not set or is empty; when it is set, it expands to empty string. That means the space will only appear when `$crossMesonFlags` exists.

Though we actually want to use `${crossMesonFlags:+$crossMesonFlags }`, since we do need the space when the variable is empty.

Here is a demonstration of how the expansions work:

```ShellSession
$ unset varname
$ echo ${varname+text}

$ echo ${varname:+text}

$ varname=
$ echo ${varname+text}
text
$ echo ${varname:+text}

$ varname=foo
$ echo ${varname+text}
text
$ echo ${varname:+text}
text
```